js/script.js

```diff
@@ -0,0 +1,106 @@
+// Интерактивная карта по клику
+
+var mapLink = document.querySelector(".contacts-and-map__image");
+
+if (mapLink) {
+  var mapPopup = document.querySelector(".popup-map");
+  var mapClose = mapPopup.querySelector(".form-close-button");
+
+  mapLink.addEventListener("click", function(evt) {
+    evt.preventDefault();
+    mapPopup.classList.add("modal-show");
+  });
+
+  mapClose.addEventListener("click", function(evt) {
+    evt.preventDefault();
+    mapPopup.classList.remove("modal-show");
+  });
+
+  window.addEventListener("keydown", function(evt) {
+    if (evt.key === "Escape") {
+      evt.preventDefault();
+      if (mapPopup.classList.contains("modal-show")) {
+        mapPopup.classList.remove("modal-show");
+      }
+    }
+  });
+}
+
+// Корзина в каталоге
+var buyButton = document.querySelectorAll(".catalog-goods-list__buy-button");
+var cart = document.querySelector(".cart");
+var buyClose = cart.querySelector(".form-close-button");
+var buyContinue = cart.querySelector(".cart__button-continue");
+
+buyButton.forEach(function(item) {
+  item.addEventListener("click", function(evt) {
+    evt.preventDefault();
+    cart.classList.add("modal-show");
+  });
+});
+
+buyClose.addEventListener("click", function(evt) {
+  evt.preventDefault();
+  cart.classList.remove("modal-show");
+});
+
+buyContinue.addEventListener("click", function(evt) {
+  evt.preventDefault();
+  cart.classList.remove("modal-show");
+});
+
+window.addEventListener("keydown", function(evt) {
+  if (evt.key === "Escape") {
+    evt.preventDefault();
+    if (cart.classList.contains("modal-show")); {
```

Comment:
```suggestion
    if (cart.classList.contains("modal-show")) {
```
лишняя точка с запятой